### PR TITLE
Refactor: revisions trees

### LINKED ISSUE #127: [Epic] Tree Listing Page
**Important**: when build status is null and we need a label, use **_Unknown_** for it

- [x] table columns like in the design: Tree - Branch - Commit/Tag (show for HEAD) - Date (for the HEAD commit) - Build Status - Boot Status - Test Status
- [x] #11 
- [x] Build status and Test status could use colors like we do in the legacy system Pass (green) /Unknown (grey) /Fail (red).
- [x] #167
- [x] #157 

obs1: Tree is identified by URL+branch

### PR DESCRIPTION
**Pull request by mari1912**
- closes #127 

image

### CODE REVIEW

**Comment by WilsonNet** [comment]
should show URL on the table because it is what define branch

File: backend/kernelCI_app/views/treeView.py [hunk 1]
```diff
@@ -14,23 +14,33 @@ def get(self, request):
...
-                        WHERE origin = %s AND start_time  >= TO_TIMESTAMP(%s)
-                        ) AS selection
-                    ORDER BY
-                        start_time DESC
-                )
+                        ORDER BY
+                            start_time DESC
+                    ) AS selection_sorted
+                WHERE
+                    origin = %s AND start_time  >= TO_TIMESTAMP(%s)
```
Comment (loadez) [inline backend/kernelCI_app/views/treeView.py:35]: I think we could move the 
WHERE
    origin = %s AND start_time  >= TO_TIMESTAMP(%s)
to inside the selection_sorted query

File: backend/requests/revision-tests-get.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-http 'http: //localhost:8000/api/revision/tests/?git_commit_hash=001f8443d480773117a013a07f774d252f369bea'
+http 'http: //localhost:8000/api/tree/tests/?git_commit_hash=001f8443d480773117a013a07f774d252f369bea'
```
Comment (WilsonNet) [inline backend/requests/revision-tests-get.sh:1]:
```suggestion
diff --git a/backend/requests/revision-tests-get.sh b/backend/requests/revision-tests-get.sh
index ad56bb9..673e759 100644
--- a/backend/requests/revision-tests-get.sh
+++ b/backend/requests/revision-tests-get.sh
@@ -1,33 +1,33 @@
-http 'http: //localhost:8000/api/tree/tests/?git_commit_hash=001f8443d480773117a013a07f774d252f369bea'
+http 'http://localhost:8000/api/tree/tests/?git_commit_hash=001f8443d480773117a013a07f774d252f369bea'
 
-//   {
-//     "current_path": "kernelci_kver",
-//     "start_time": "2024-08-08T21:46:15.254000Z",
-//     "fail_tests": 0,
-//     "error_tests": 0,
-//     "miss_tests": 0,
-//     "pass_tests": 1,
-//     "done_tests": 0,
-//     "skip_tests": 0,
-//     "null_tests": 0,
-//     "total_tests": 1,
-//     "origins": [
-//       "maestro"
-//     ]
-//   },
-//   {
-//     "current_path": "kunit",
-//     "start_time": "2024-08-08T21:46:24.509000Z",
-//     "fail_tests": 0,
-//     "error_tests": 0,
-//     "miss_tests": 0,
-//     "pass_tests": 327,
-//     "done_tests": 0,
-//     "skip_tests": 44,
-//     "null_tests": 0,
-//     "total_tests": 371,
-//     "origins": [
-//       "maestro"
-//     ]
-//   }
-// ]
\ No newline at end of file
+#   {
+#     "current_path": "kernelci_kver",
+#     "start_time": "2024-08-08T21:46:15.254000Z",
+#     "fail_tests": 0,
+#     "error_tests": 0,
+#     "miss_tests": 0,
+#     "pass_tests": 1,
+#     "done_tests": 0,
+#     "skip_tests": 0,
+#     "null_tests": 0,
+#     "total_tests": 1,
+#     "origins": [
+#       "maestro"
+#     ]
+#   },
+#   {
+#     "current_path": "kunit",
+#     "start_time": "2024-08-08T21:46:24.509000Z",
+#     "fail_tests": 0,
+#     "error_tests": 0,
+#     "miss_tests": 0,
+#     "pass_tests": 327,
+#     "done_tests": 0,
+#     "skip_tests": 44,
+#     "null_tests": 0,
+#     "total_tests": 371,
+#     "origins": [
+#       "maestro"
+#     ]
+#   }
+# ]

```

File: dashboard/src/components/Table/TreeTable.tsx [hunk 3]
```diff
@@ -50,7 +52,9 @@ const TreeTableRow = (row: TreeTableBody): JSX.Element => {
       <TableCell>
         {sanitizeTableValue(row.tree_names.join(', '), false)}
       </TableCell>
+      <TableCell>{sanitizeTableValue(row.branch, false)}</TableCell>
       <TableCell>{sanitizeTableValue(row.commit)}</TableCell>
+      <TableCell>{sanitizeTableValue(row.date.split('T')[0])}</TableCell>
```
Comment (WilsonNet) [inline dashboard/src/components/Table/TreeTable.tsx:57]: Array indexes on typescript can return undefined and the linter won't warn us unless we have a rule for that, maybe add a validation here

**Review by WilsonNet — DISMISSED** [review]
nice job!!!

**Review by loadez — APPROVED** [review]

**Review by WilsonNet — APPROVED** [review]
n1++++

**Pull request merged**